lib/constructs/interface.js

```diff
@@ -26,6 +26,35 @@ Interface.prototype.implements = function (source) {
   this.mixins.push(source);
 };
 
+Interface.prototype.generateNamedPropertiesObject = function () {
+  if (!utils.isGlobal(this.idl)) {
+    return;
+  }
+
+  this.str += `var ${this.name}PropertiesConstructor = function () {
```

Comment:
You can do `window.__proto__.__proto__.constructor` and it'll return a function which throws `Illegal constructor`.